lib/plug/cowboy/conn.ex

```diff
@@ -90,6 +90,10 @@ defmodule Plug.Cowboy.Conn do
     :cowboy_req.inform(status, to_headers_map(headers), req)
   end
 
+  @impl true
+  def upgrade(req, :websocket, args), do: {:ok, Map.put(req, :upgrade, {:websocket, args})}
```

Comment:
We should probably validate here that `args` is a triplet, the first element is an atom and the third is a map.

Reply:
I was thinking about that at the dog park yesterday - maybe we should add a matching pattern match in the plug PR?

Reply:
I think it is fine to raise here?

Reply:
Updated to explicitly match on a three element tuple